File: apps/java/io/dustin/apps/board/domain/restriction/posting/service/WritePostingRestrictionService.java
package io.dustin.apps.board.domain.restriction.posting.service;

import io.dustin.apps.board.domain.restriction.posting.model.PostingRestriction;
import io.dustin.apps.board.domain.restriction.posting.repository.PostingRestrictionRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor
@Service
public class WritePostingRestrictionService {

    private final PostingRestrictionRepository postingRestrictionRepository;

    public PostingRestriction create(Long fromUserId, Long toUserId, Long postingId){
        PostingRestriction postingRestriction = PostingRestriction.builder()
                .fromUserId(fromUserId)
                .toUserId(toUserId)
                .postingId(postingId)
                .build();
        this.postingRestrictionRepository.save(postingRestriction);
        return postingRestriction;
    }

    public void delete(Long fromUserId, Long toUserId, Long postingId) { postingRestrictionRepository.deleteByFromUserIdAndToUserIdAndPostingId(fromUserId, toUserId, postingId); }
}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/restriction/posting/service/WritePostingRestrictionService.kt
package io.dustin.apps.board.domain.restriction.posting.service

import io.dustin.apps.board.domain.restriction.posting.model.PostingRestriction
import io.dustin.apps.board.domain.restriction.posting.repository.PostingRestrictionRepository
import org.springframework.stereotype.Service

@Service
class WritePostingRestrictionService(
    private val postingRestrictionRepository: PostingRestrictionRepository
) {

    /**
     * 유저 제한
     */
    fun create(fromUserId: Long, toUserId: Long, postingId: Long): PostingRestriction {
        val postingRestriction = PostingRestriction(
            fromUserId = fromUserId,
            toUserId =  toUserId,
            postingId = postingId,
        )
        postingRestrictionRepository.save(postingRestriction)
        return postingRestriction
    }

    /**
     * 유저 제한 해제
     */
    fun delete(fromUserId: Long, toUserId: Long, postingId: Long) {
        postingRestrictionRepository.deleteByUsers(fromUserId, toUserId, postingId)
    }
}
